src/Raven.Client/Documents/Session/AbstractDocumentQuery.Sharding.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Linq;
+using Raven.Client.Documents.Session.Querying.Sharding;
+
+namespace Raven.Client.Documents.Session;
+
+public abstract partial class AbstractDocumentQuery<T, TSelf>
+{
+    protected void ShardContext(Action<IShardedQueryContextBuilder> builder)
+    {
+        var builderImpl = new ShardedQueryContextBuilder();
+
+        builder.Invoke(builderImpl);
+
+        object shardContext;
+
+        if (builderImpl.DocumentIds.Count == 1)
+            shardContext = builderImpl.DocumentIds.First();
+        else
+            shardContext = builderImpl.DocumentIds;
```

Comment:
Why do this? We can just pass an array in all cases.

Reply:
We can do it. I just thought that since we have the distinction of the parameter type on the server side:

https://github.com/ravendb/ravendb/blob/dcf9e05775a18de4a56f912621f453a089d4da15/src/Raven.Server/Documents/Sharding/Queries/ShardedQueryProcessor.cs#L58-L61

we'll take advantage of that on the client API as well. At least we'll test that code properly.

Reply:
I'm fine either way